src/Prometheus.Client.HttpRequestDurations/Tools/HttpContextExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Microsoft.AspNetCore.Http;
+#if HasRoutes
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Routing;
+#endif
+
+namespace Prometheus.Client.HttpRequestDurations.Tools
+{
+    internal static class HttpContextExtensions
+    {
+#if HasRoutes
+        public static string GetRouteName(this HttpContext httpContext)
```

Comment:
Need some examples with result. Maybe will you add unit tests?

Reply:
I'm not sure how to mock the http url parsing by ASP NET router. I'll try to find out what can we do here.